marklogic-client-api/src/main/java/com/marklogic/client/datamovement/LineSplitter.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2019 MarkLogic Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.marklogic.client.datamovement;
+
+import com.marklogic.client.io.Format;
+import com.marklogic.client.io.StringHandle;
+
+import java.io.*;
+import java.nio.charset.Charset;
+import java.util.stream.Stream;
+
+/**
+ * java doc to do
+ */
+public class LineSplitter implements Splitter<StringHandle> {
+    private Format format = Format.JSON;
+    private int count = 0;
+
+    /**
+     * Returns the document format set to splitter.
+     * @return the document format set to splitter. The default is Format.JSON.
+     */
+    public Format getFormat() {
+        return this.format;
+    }
+
+    /**
+     * Used to set document format to splitter
+     * @param format the document content format.
+     */
+    public void setFormat(Format format) {
+        if (format == null) {
+            throw new IllegalArgumentException("some exception");
+        }
+
+        this.format = format;
+    }
+
+    /**
+     *
+     * @return the number of objects in the stream.
+     */
+    @Override
+    public long getCount() {
+        return count;
+    }
+
+    /**
+     * Takes the input stream and converts it into a stream of StringHandle. The content could be
+     * line-delimited JSON file, line-delimited XML file or gzip-compressed line-delimited JSON file.
```

Comment:
Would it be useful to specifically mention a GZIPInputStream?  Also, wouldn't compression work for XML as well as JSON?

Reply:
It should work with gzipped XML File. I changed the description as below.

     * Takes the input stream and converts it into a stream of StringHandle. The content could be
     * line-delimited JSON, XML or TEXT file. It could also be gzip-compressed line-delimited files.
     * Provide GZIPInputStream to the splitter when splitting gzip files.